### ISSUE TITLE
Use different kind of wrappers in the same experiment (or at least have different wrappers of the same kind with different jobs)

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#420]
In GitLab by @mcastril on Jul 16, 2019, 12:55

**Comment by kinow** [comment]
In GitLab by @mcastril on Jan 13, 2021, 12:37

I think we could think in supporting this feature before the transition, maybe for 3.14. As with #335, it depends on how much disruptive it would be. But this is a very useful feature and I think we won't have AS4 in production before summer.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jan 13, 2021, 12:59

Okay, you're right. 

I'll implement it then

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jul 22, 2021, 13:13

This feature is now fully merged into 3.14.0b(not yet on modules), I'm doing some changes to the inner jobs #700. 

So I'll need to test whenever the maintenance ends

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 23, 2021, 14:56

There were fixes while working on other branches that affected this issue. I need to run tests with real templates on devel now that all is merged into master

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 23, 2021, 17:58

One good example is the `Auto-EC-Earth` case. You can create a vertical wrapper for the SIMs and a horizontal wrapper to join all the CMOR jobs together. It would be nice to distribute them equally by socket... 1 and 1, 2 and 2...

I think that your test only contains CMORATM and CMOROCE. You can start testing that one.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jan 13, 2022, 12:52

I found some bugs regarding wrapper that are also present in 3.13.0 although it won't affect the wrapper generation on 3.13.0. 

In case, that a job is a substring of another job, it will be added to the one of the dicts that autosubmit uses for creating the workflow. 

Example: SIM and SIM2. SIM will be added into the dict of SIM2 which caused issues with multiwrapper mode.

This has been solved and tested today on 3.14.0.
